src/components/Icon/Icon.tsx

```diff
@@ -213,19 +215,42 @@ function Icon({
     backdrop && styles.hasBackdrop,
   );
 
+  const defaultIconProps = {
+    className: styles.Svg,
+    focusable: 'false',
+    'aria-hidden': 'true',
+  };
+
+  if (untrusted) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'Deprecation: The untrusted prop is no longer needed, all strings passed into the Icon component are rendered as unsafe',
+    );
+  }
+
+  let SourceComponent = source as React.SFC<React.SVGProps<SVGSVGElement>>;
   let contentMarkup: React.ReactNode;
   if (source === 'placeholder') {
     contentMarkup = <div className={styles.Placeholder} />;
+  } else if (isBundledIcon(source)) {
+    SourceComponent = BUNDLED_ICONS[source];
+    contentMarkup = <SourceComponent {...defaultIconProps} />;
+  } else if (typeof source === 'function') {
+    const sourceElement = <SourceComponent {...defaultIconProps} />;
+    if (React.isValidElement(sourceElement)) {
```

Comment:
I do want to keep the React.isValidElement check in there.

Right now, the Icon component just fails silently if you pass in something that it doesn't understand. I want to change that in a follow-up PR and make it throw an exception, but without the isValidElement check in there this would accept any function and not be as useful.